### PR TITLE
 Add Get/Update methods for Properties API

### PR DESCRIPTION
**Pull request by abhide**

### CODE REVIEW

File: clients/splunk/properties.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	return json.Unmarshal(body, v)
+}
+
+// UpdateKey updates value for specified key from the specified stanza in the configuration file
+func (p *PropertiesService) UpdateKey(file string, stanza string, key string, value string) (*string, *http.Response, error) {
+	apiError := &APIError{}
+
+	body := strings.NewReader(fmt.Sprintf("value=%s", value))
+	resp, err := p.client.New().Post(
+			fmt.Sprintf("properties/%s/%s/%s", file, stanza, key)).Body(body).ResponseDecoder(stringResponseDecoder{}).Receive(nil, apiError)
```
Comment (michaelw) [inline clients/splunk/properties.go]: do the interpolated values need some kind of escaping?

File: clients/splunk/properties.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	}
+	return nil, resp, relevantError(err, apiError)
+}
+
+// GetKey returns value for the given key from the specified stanza in the configuration file
+func (p *PropertiesService) GetKey(file string, stanza string, key string) (*string, *http.Response, error) {
+	apiError := &APIError{}
+	output := &Entry{}
+	resp, err := p.client.New().Get(
+		fmt.Sprintf("properties/%s/%s/%s", file, stanza, key)).ResponseDecoder(stringResponseDecoder{}).Receive(output, apiError)
```
Comment (michaelw) [inline clients/splunk/properties.go]: ditto, see above

File: clients/splunk/properties.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	if err != nil || !apiError.Empty() {
```
Comment (michaelw) [inline clients/splunk/properties.go:73]: hmm, maybe we should make it such that we can check for exactly one condition.

File: clients/splunk/properties.go (hunk 1 above)
Comment (abhide) [reply clients/splunk/properties.go]: Fixed

File: clients/splunk/properties.go (hunk 2 above)
Comment (abhide) [reply clients/splunk/properties.go]: Fixed

File: clients/splunk/properties.go (hunk 3 above)
Comment (abhide) [reply clients/splunk/properties.go:73]: Will address these in a cleanup/refactor separate PR.

File: Makefile [hunk 4]
```diff
@@ -31,18 +26,12 @@ dev: build
 test: build
 	@test -n "$$SPLUNK_ADDR" || { echo 'warning: SPLUNK_ADDR not set, creating new Splunk instances.  This will be slow.'; }
 	mkdir -p $(dir $(TESTREPORT))
-	gotestsum --junitfile $(TESTREPORT) -- -cover -v ./...
+	go clean -testcache || true
```
Comment (michaelw) [inline Makefile:29]: ```env GOCACHE=off gotestsum...```

Comment (abhide) [reply Makefile:29]:
```$ make test
mkdir -p ./
env GOCACHE=off gotestsum --junitfile test-results.xml --format standard-verbose -- -cover -v ./...
build cache is disabled by GOCACHE=off, but required as of Go 1.12

=== Errors
build cache is disabled by GOCACHE=off, but required as of Go 1.12

DONE 0 tests, 1 error in 0.015s
make: *** [test] Error 1```

**Pull request merged**